Entity/Document.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+
+namespace EMS\CommonBundle\Entity;
+
+
+class Document
+{
+    /**@var string */
+    private $contentType;
+    /**@var string */
+    private $ouuid;
+    /**@var array */
+    private $source;
+
+    public function __construct(string $contentType, string $ouuid, array $source)
+    {
+        $this->contentType = $contentType;
+        $this->ouuid = $ouuid;
+        $this->source = $source;
+    }
+
+    /**
+     * @return string
+     */
+    public function getContentType(): string
+    {
+        return $this->contentType;
+    }
+
+    /**
+     * @param string $contentType
+     * @return Document
+     */
+    public function setContentType(string $contentType): Document
```

Comment:
remove the setters. You should not be allowed to modify a Document that comes from a search operation. That would alter the reality. 

If one really needs a modification in a Document, they should create a new Document with the values of the old one that they want to keep, and pass the modification via the constructor

Reply:
Agree

Reply:
Done